neuro_san/session/session_invocation_context.py

```diff
@@ -57,30 +58,35 @@ def __init__(self, async_session_factory: AsyncAgentSessionFactory,
         """
         Constructor
 
+        :param agent_name: The name of the agent
         :param async_session_factory: The AsyncAgentSessionFactory to use
                         when connecting with external agents.
         :param async_executors_pool: pool of AsyncioExecutors to use for obtaining
                          an executor instance to use for this context;
         :param llm_factory: The ContextTypeLlmFactory instance
         :param toolbox_factory: The ContextTypeToolboxFactory instance
-        :param metadata: A grpc metadata of key/value pairs to be inserted into
+        :param metadata: A request metadata of key/value pairs to be inserted into
                          the header. Default is None. Preferred format is a
                          dictionary of string keys to string values.
         :param reservationist: The Reservationist instance to use.
         """
 
+        # From args
+        self.agent_name: str = agent_name
         self.async_session_factory: AsyncAgentSessionFactory = async_session_factory
         self.async_executors_pool: AsyncioExecutorPool = async_executors_pool
+        self.llm_factory: ContextTypeLlmFactory = llm_factory
+        self.toolbox_factory: ContextTypeToolboxFactory = toolbox_factory
+        self.metadata: Dict[str, str] = metadata
+        self.reservationist: Reservationist = reservationist
```

Comment:
A little grouping of the member variables so it's clearer what comes from the args and what doesn't.
Also add the agent_name so we can report it. Can't believe we didn't have this before.